src/snowflake/snowpark/mock/_functions.py

```diff
@@ -822,7 +823,7 @@ def mock_timestamp_ntz(
     result = _to_timestamp(column, fmt, try_cast, enforce_ltz=True)
     # Cast to NTZ by removing tz data if present
     return ColumnEmulator(
-        data=[x.replace(tzinfo=None) for x in result],
+        data=[None if x is None else x.replace(tzinfo=None) for x in result],
```

Comment:
NIT: can we rewrite this to use `try_convert`?

Reply:
Yes, but it's a bit more verbose since I'd have to use a lambda function.